src/test/clojure/clara/test_rules.clj

```diff
@@ -5025,3 +5029,33 @@
     (is (= (query end-session-equal-facts lousy-weather-query)
            (query end-session-identical-facts lousy-weather-query)
            [{}]))))
+
+(deftest test-alpha-batching-with-multiple-ancestors
+  ;; Validate that two distinct fact types are batched together when passed to a condition on a
+  ;; common ancestor type. We use an accumulator because it provides a convenient way to directly
+  ;; access the facts passed to the condition without exposing Clara internals.
+  ;; see issue 257 for more details of the bug surrounding this test
+  (let [accum-state (atom #{})
+        
+        stateful-accum (acc/accum
+                         {:initial-value []
+                          :reduce-fn conj
+                          :retract-fn (fn [items retracted] (remove #{retracted} items))
+                          :convert-return-fn (fn [items] 
+                                               (do 
+                                                 (swap! accum-state conj items)
+                                                 items))})
+        
+        common-ancestor-rule (dsl/parse-rule [[?maps <- stateful-accum :from [List]]]
```

Comment:
Could you name this something other than ?maps please?  Another leftover from your previous version where these were maps I presume.